clang/lib/AST/ByteCode/Compiler.cpp

```diff
@@ -1300,6 +1309,16 @@ bool Compiler<Emitter>::VisitVectorBinOp(const BinaryOperator *E) {
       if (!this->emitGT(ElemT, E))
         return false;
       break;
+    case BO_LAnd:
+      // a && b is equivalent to a!=0 & b!=0
```

Comment:
I was trying to prove this wrong but it's really what the current interpreter does. Since the vector isn't implicitly convertible to bool, there's no short-circuiting: https://godbolt.org/z/aaoTPWcKo

Reply:
Should we follow the behavior of current interpreter or go further to implement a short-circuiting here? 
I have go through the Clang issue list, and it seems that no users have reported related bugs.

- https://github.com/llvm/llvm-project/issues?q=is%3Aissue+is%3Aopen+vector_size+label%3Aclang%3Afrontend
- https://github.com/llvm/llvm-project/issues?q=is%3Aissue+is%3Aopen+ext_vector_type+label%3Aclang%3Afrontend

Reply:
Not sure how one would implement short-circuiting if a vector can't be converted to a bool. Following what the current interpreter does is fine.